### PR TITLE
Devise Integration

### PR DESCRIPTION
**Pull request by AseelaAbdulSattar**
Please review "Devise Integration"

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -50,5 +50,7 @@ group :test do
   gem 'webdrivers'
 end
 
+#Adding gem devise
```
Comment (hassanakram) [inline Gemfile]: Don't add comments unless very necessary to explain a complex code block.

File: app/assets/stylesheets/application.css [hunk 2]
```diff
@@ -13,3 +13,6 @@
  *= require_tree .
  *= require_self
  */
+ h1 {
+  text-align: center;
+}
```
Comment (hassanakram) [inline app/assets/stylesheets/application.css]: There should be end of line in each file, check VSCode settings, you can enable it on save.

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,2 +1,15 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
```
Comment (hassanakram) [inline app/controllers/application_controller.rb]: Authentication filter should be added to only those controllers in which we want user to be signed in. Adding it in application controller will make the whole application closed to signed in users only.

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -10,6 +10,23 @@
   </head>
 
   <body>
+  <p class="navbar-text float-right">
```
Comment (hassanakram) [inline app/views/layouts/application.html.erb]: wrong indentation, this should never happen.

File: config/environments/development.rb [hunk 5]
```diff
@@ -59,4 +59,7 @@
   # Use an evented file watcher to asynchronously detect changes in source code,
   # routes, locales, etc. This feature depends on the listen gem.
   config.file_watcher = ActiveSupport::EventedFileUpdateChecker
+
+  #Added
```
Comment (hassanakram) [inline config/environments/development.rb]: remove this unnecessary comment.

File: config/initializers/devise.rb [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+
+  # ==> Controller configuration
+  # Configure the parent class to the devise controllers.
+  # config.parent_controller = 'DeviseController'
+
+  # ==> Mailer Configuration
+  # Configure the e-mail address which will be shown in Devise::Mailer,
+  # note that it will be overwritten if you use your own mailer class
+  # with default "from" parameter.
+  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
```
Comment (hassanakram) [inline config/initializers/devise.rb]: change this to a meaningful from email, i.e info@fakebook.com

File: app/controllers/application_controller.rb (hunk 3 above)
Comment (AseelaAbdulSattar) [reply app/controllers/application_controller.rb]: a user always need to signed in e.g, to like post or comment on a post or send a friend request etc.  that's why i put it in application controller.

File: app/views/layouts/application.html.erb (hunk 4 above)
Comment (AseelaAbdulSattar) [reply app/views/layouts/application.html.erb]: Updated

**Pull request merged**